core/src/main/scala/com/avast/dapper/DefaultCassandraDao.scala

```diff
@@ -0,0 +1,76 @@
+package com.avast.dapper
+
+import java.util.concurrent.Executor
+
+import com.datastax.driver.core._
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DefaultCassandraDao[PrimaryKey, Entity <: CassandraEntity[PrimaryKey]](session: Session)(
+    implicit entityMapper: EntityMapper[PrimaryKey, Entity],
+    ec: ExecutionContext,
```

Comment:
Can't you use `ExecutionContextExecutor`?

Reply:
It originally was like that. The problem is the `Executor` is used for conversion between `ListenableFuture` and `Future` and you remember how it works. So this split was about "think twice before you use your callback executor to this".

Reply:
Then the `Executor` should not be `implicit`. First of all I think it could cause problems because regularly `ExecutionContext.global` is both `ExecutionContext` and `Executor` so you would have to manually fix it.  Second of all I don't think pools for blocking calls should be implicit.